nixos/modules/virtualisation/xen-dom0.nix

```
@@ -1,213 +1,907 @@
+                  example = true;
+                  type = lib.types.bool;
+                  visible = false;
+                  description = "Set `access-log-special-ops`.";
+                };
+              };
+              xenfs = {
+                kva = lib.options.mkOption {
+                  default = "/proc/xen/xsd_kva";
+                  example = cfg.store.settings.xenstored.xenfs.kva;
+                  type = lib.types.path;
+                  visible = false;
+                  description = ''
+                    Path to the Xen Store Daemon KVA location inside the XenFS pseudo-filesystem.
+                    While it is possible to alter this value, some drivers may be hardcoded to follow the default paths.
+                  '';
+                };
+                port = lib.options.mkOption {
+                  default = "/proc/xen/xsd_port";
+                  example = cfg.store.settings.xenstored.xenfs.port;
+                  type = lib.types.path;
+                  visible = false;
+                  description = ''
+                    Path to the Xen Store Daemon userspace port inside the XenFS pseudo-filesystem.
+                    While it is possible to alter this value, some drivers may be hardcoded to follow the default paths.
+                  '';
+                };
+              };
+            };
+            ringScanInterval = lib.options.mkOption {
+              default = 20;
+              example = 30;
+              type = lib.types.addCheck (
+                lib.types.int
+                // {
+                  name = "nonzeroInt";
+                  description = "nonzero signed integer, meaning !=0";
+                  descriptionClass = "nonRestrictiveClause";
+                }
+              ) (n: n != 0);
+              description = ''
+                Perodic scanning for all the rings as a safenet for lazy clients.
+                Define the interval in seconds; set to a negative integer to disable.
+              '';
+            };
+          };
+        };
+      };
+    };
   };
 
+  ## Implementation ##
 
-  ###### implementation
-
-  config = mkIf cfg.enable {
-    assertions = [ {
-      assertion = pkgs.stdenv.isx86_64;
-      message = "Xen currently not supported on ${pkgs.stdenv.hostPlatform.system}";
-    } {
-      assertion = config.boot.loader.grub.enable && (config.boot.loader.grub.efiSupport == false);
-      message = "Xen currently does not support EFI boot";
-    } ];
-
-    virtualisation.xen.package = mkDefault pkgs.xen;
-    virtualisation.xen.package-qemu = mkDefault pkgs.xen;
-    virtualisation.xen.stored = mkDefault "${cfg.package}/bin/oxenstored";
+  config = lib.modules.mkIf cfg.enable {
+    assertions = [
+      {
+        assertion = pkgs.stdenv.isx86_64;
+        message = "Xen is currently not supported on ${pkgs.stdenv.hostPlatform.system}.";
+      }
+      {
+        assertion = config.boot.loader.systemd-boot.enable;
```

```suggestion
        assertion = config.boot.loader.systemd-boot.enable || (config.boot ? lanzaboote) && config.boot.lanzaboote.enable;
```

It is not terribly broken on lanzaboote, It just boots NixOS without Xen, adding this here to remove that change from my nixpkgs fork, this is the only change needed to make upstream lanzaboote nixosModule from my branch work (https://github.com/nix-community/lanzaboote/pull/387).